libcnb/src/build.rs

```diff
@@ -27,6 +19,21 @@ pub struct BuildContext<B: Buildpack + ?Sized> {
     pub buildpack_descriptor: BuildpackToml<B::Metadata>,
 }
 
+impl<B: Buildpack + ?Sized> BuildContext<B> {
+    pub fn handle_layer<L: Layer<Buildpack = B>>(
+        &self,
+        name: impl AsRef<str>,
+        layer: L,
+    ) -> crate::Result<LayerData<L::Metadata>, B::Error> {
+        crate::layer::handle_layer(self, name.as_ref(), layer).map_err(|error| match error {
+            HandleLayerErrorOrBuildpackError::HandleLayerError(e) => {
+                crate::Error::HandleLayerError(e)
+            }
+            HandleLayerErrorOrBuildpackError::BuildpackError(e) => crate::Error::BuildpackError(e),
+        })
+    }
+}
+
```

Comment:
`handle_layer` needs tests for sure. It's a bit tricky right now since it requires a `BuildContext` with all the related values in it. I did forgo tests for it at this stage and will submit another PR that improves testability by providing "standard" values for things like `BuildpackToml` and similar. Are you okay with that approach? Tracking tests for `handle_layer` here: https://github.com/Malax/libcnb.rs/issues/169

Reply:
Thanks 🙏 I think this was my last blocker